pingcap/tidb
# log: desensitize of dispatching error and parsing error

### What problem does this PR solve?

Issue Number: close #21128 

Problem Summary:

### What is changed and how it works?

What's Changed:

- `syntax error` now becomes debug level error
- `parse SQL failed` will become debug level error when enable redact log

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

1. not enable tidb_redact_log

```
mysql> alter user set password 'aaae';
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 14 near "set password 'aaae'"

[2020/11/19 14:43:39.620 +08:00] [WARN] [session.go:1175] ["parse SQL failed"] [conn=6879507915552587783] [error="line 1 column 14 near \"set password 'aaae'\" "] [SQL="alter user set password 'aaae'"]
[2020/11/19 14:43:39.620 +08:00] [INFO] [conn.go:801] ["command dispatched failed"] [conn=6879507915552587783] [connInfo="id:6879507915552587783, addr:127.0.0.1:50671 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="alter user set password 'aaae'"] [txn_mode=PESSIMISTIC] [err="[parser:1064]You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 14 near \"set password 'aaae'\" "]
```

2. enable tidb_redact_log

```
mysql> set @@session.tidb_redact_log=on;
Query OK, 0 rows affected (0.01 sec)
mysql> alter user set password 'aaae';
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 14 near "set password 'aaae'"

[2020/11/19 15:21:22.241 +08:00] [INFO] [conn.go:801] ["command dispatched failed"] [conn=2405946945852932103] [connInfo="id:2405946945852932103, addr:127.0.0.1:57504 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="alter user set password ?"] [txn_mode=PESSIMISTIC] [err="fail to parse SQL and can't desensitize when enable log redaction"]
```

3. BRIE SQL

```
mysql> RESTORE DATABASE *     FROM 's3://storage/url?access-key=AAAAA&secret-access-key=SSSSS&force-path-style=1&endpoint=http://10.0.1.55:8888'     RATE_LIMIT = 512 MB/SECOND;
ERROR 8125 (HY000): Restore failed: Bucket storage is not accessible: RequestError: send request failed
caused by: Head http://10.0.1.55:8888/storage: dial tcp 10.0.1.55:8888: i/o timeout: [BR:ExternalStorage:ErrStorageInvalidConfig]invalid external storage config

[2020/11/19 14:59:14.800 +08:00] [INFO] [conn.go:801] ["command dispatched failed"] [conn=1143914305352105987] [connInfo="id:1143914305352105987, addr:127.0.0.1:53432 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="restore database * from ? rate_limit = ? mb / second"] [txn_mode=PESSIMISTIC] [err="[executor:8125]Restore failed: Bucket storage is not accessible: RequestError: send request failed\ncaused by: Head http://10.0.1.55:8888/storage: dial tcp 10.0.1.55:8888: i/o timeout: [BR:ExternalStorage:ErrStorageInvalidConfig]invalid external storage config\ngithub.com/pingcap/errors.AddStack\n\t/Users/lance/Projects/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201029093017-5a7df2af2ac7/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\t/Users/lance/Projects/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201029093017-5a7df2af2ac7/normalize.go:156\ngithub.com/pingcap/tidb/executor.handleBRIEError\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/executor/brie.go:366\ngithub.com/pingcap/tidb/executor.(*BRIEExec).Next\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/executor/brie.go:345\ngithub.com/pingcap/tidb/executor.Next\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/executor/executor.go:268\ngithub.com/pingcap/tidb/executor.(*recordSet).Next\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/executor/adapter.go:128\ngithub.com/pingcap/tidb/server.(*tidbResultSet).Next\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/server/driver_tidb.go:284\ngithub.com/pingcap/tidb/server.(*clientConn).writeChunks\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/server/conn.go:1713\ngithub.com/pingcap/tidb/server.(*clientConn).writeResultset\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/server/conn.go:1673\ngithub.com/pingcap/tidb/server.(*clientConn).handleStmt\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/server/conn.go:1570\ngithub.com/pingcap/tidb/server.(*clientConn).handleQuery\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/server/conn.go:1436\ngithub.com/pingcap/tidb/server.(*clientConn).dispatch\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/server/conn.go:1012\ngithub.com/pingcap/tidb/server.(*clientConn).Run\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/server/conn.go:784\ngithub.com/pingcap/tidb/server.(*Server).onConn\n\t/Users/lance/Projects/go/src/github.com/pingcap/tidb/server/server.go:459\nruntime.goexit\n\t/usr/local/Cellar/go/1.13.6/libexec/src/runtime/asm_amd64.s:1357"]
```

Side effects


### Release note 

- No release note
